packages/cadl-ranch-specs/http/models/property-optional/mockapi.ts

```diff
@@ -33,9 +36,8 @@ function createMockApis(route: string, value: any, convertBodyProperty?: (_: any
     };
   });
   const putAll = mockapi.put(allUrl, (req) => {
-    if (convertBodyProperty && req.originalRequest.body?.property && allBody?.property) {
+    if (convertBodyProperty && req.originalRequest.body?.propertyy) {
       req.originalRequest.body.property = convertBodyProperty(req.originalRequest.body.property);
-      allBody.property = convertBodyProperty(allBody.property);
     }
 
     req.expect.bodyEquals(allBody);
```

Comment:
I think we should
- If there is a convertBodyProperty, create a new local allBody with converted body and `req.expect.bodyEquals(localAllBody);`
- Otherwise still `req.expect.bodyEquals(allBody);`

Reply:
Hi @pshao25, 

I added code here: https://github.com/Azure/cadl-ranch/pull/153/files#diff-7cedfaa56d319065d6385f01969648a56e8d91556bfa0b9018a99ad9b0e4a2bdR23 ,
It can achieve same goal as your description?

Reply:
No, I hope the expected value `allBody` not impacted, so that e.g. when we call `get`, we could still get the unmodified value.

Reply:
I'm not sure this is the right direction.
It will make the data value in wire of `put` and `get` be different for `datetime` while other datetypes have exactly the same values between these two operations.

Reply:
Hi @pshao25 , I updated as offline discussion.

Reply:
I’m going to go ahead and merge thanks changlong